src/main/resources/META-INF/rewrite/java-ee-8.yml

```diff
@@ -0,0 +1,61 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.javaee8
+displayName: Migrate to JavaEE8
+description: These recipes help with the Migration to Java EE 8, flagging and updating deprecated methods.
+tags:
+  - javaee8
+  - deprecated
+recipeList:
+  - org.openrewrite.java.migrate.javaee7
+  - org.openrewrite.java.migrate.javaee8.ServletIsRequestedSessionIdFromURL
+  - org.openrewrite.java.migrate.javaee8.ApacheDefaultProvider
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.javaee8.ServletIsRequestedSessionIdFromURL
+displayName: Replace `HttpServletRequestWrapper.isRequestedSessionIdFromUrl()` with `HttpServletRequestWrapper.isRequestedSessionIdFromURL()`
+description: The  method `HttpServletRequestWrapper.isRequestedSessionIdFromUrl()` is deprecated in JavaEE8 and is replaced by `HttpServletRequestWrapper.isRequestedSessionIdFromURL()`
+recipeList:
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: javax.servlet.http.HttpServletRequestWrapper isRequestedSessionIdFromUrl(*)
+      newMethodName: isRequestedSessionIdFromURL
+      ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.javaee8.ApacheDefaultProvider
+displayName: Flags any `org.apache.bval.jsr*` (bval 1.1) and `org.apache.bval.jsr303*` (bval 1.0) package references
+description: >
+  This recipe flags any org.apache.bval.jsr* (bval 1.1) and org.apache.bval.jsr303* (bval 1.0) package references in validation.xml deployment descriptors. 
+  Bean Validation 2.0 and later use the Hibernate Validator implementation instead of the 
+  Apache BVal implementation which was used for Bean Validation 1.0 and 1.1
+recipeList:
+  - org.openrewrite.xml.ChangeTagValue:
+      elementName: //default-provider
+      newValue: org.hibernate.validator.HibernateValidator
+  - org.openrewrite.xml.ChangeTagValue:
+      elementName: //message-interpolator
+      newValue: org.hibernate.validator.messageinterpolation.ResourceBundleMessageInterpolator
+  - org.openrewrite.xml.ChangeTagValue:
+      elementName: //traversable-resolver
+      newValue: org.hibernate.validator.engine.resolver.DefaultTraversableResolver
+  - org.openrewrite.xml.ChangeTagValue:
+      elementName: //constraint-validator-factory
+      newValue: org.hibernate.validator.engine.ConstraintValidatorFactoryImpl
+  - org.openrewrite.xml.RemoveXmlTag:
+      xPath: //parameter-name-provider
```

Comment:
Same as above: please use the full xpath expressions such that we can more safely run against projects that might contain other XML files with similarly named elements.

Reply:
done